packages/tests/server/react/useQueries.test.tsx

```diff
@@ -0,0 +1,59 @@
+import { getServerAndReactClient } from './__reactHelpers';
+import { render, waitFor } from '@testing-library/react';
+import { initTRPC } from '@trpc/server/src';
+import { konn } from 'konn';
+import React from 'react';
+import { z } from 'zod';
+
+const ctx = konn()
+  .beforeEach(() => {
+    const t = initTRPC.create({
+      errorFormatter({ shape }) {
+        return {
+          ...shape,
+          data: {
+            ...shape.data,
+            foo: 'bar' as const,
+          },
+        };
+      },
+    });
+    const appRouter = t.router({
+      post: t.router({
+        byId: t.procedure
+          .input(
+            z.object({
+              id: z.string(),
+            }),
+          )
+          .query(() => '__result' as const),
+      }),
+    });
+
+    return getServerAndReactClient(appRouter);
+  })
+  .afterEach(async (ctx) => {
+    await ctx?.close?.();
+  })
+  .done();
+
+test('useQueries()', async () => {
+  const { proxy, App } = ctx;
+  function MyComponent() {
+    const utils = proxy.useContext();
+    const results = proxy.useQueries({
+      queries: [utils.post.byId.getQueryOptions({ id: '1' })],
+    });
```

Comment:
WDYT about this as an alternative API? Feels more intuitive to me

```suggestion
const results = proxy.useQueries(q => [ q.post.byId({ id: "1" }) ])
```

Reply:
on second look, maybe it's not useful anywhere other than `useQueries`, cc @TkDodo

Reply:
if we expose the options, I think it's weird that we expose the queryKey separately as I've mentioned here:
- https://github.com/trpc/trpc/pull/3305#discussion_r1033190006

I also like the alternative api, and yes, I also wouldn't know what we would need the other options for _apart from_ the queryKey

Reply:
Ok yeah I think I'll go for both here then. I'll keep `getQueryOptions` and we probably not do `getQueryKey` then? And I'll also do the special proxy for `useQueries`

Reply:
Just my 2 cents... I actually think not exposing a query options object on the `useContext()` object and letting Julius get to the bottom of how best to implement a getQueryKey function is a better idea; especially as he is handling router level as well as queries at a leaf node (ie `post` router as a whole not just `post.getById` )

Also I think the alternative api of a seperate proxy provided as an input to the use Queries function is better too (the afore called alternative api 🙂)

Reply:
We could do like

- on trpc object, expose getQueryKey. This would be callable outside of the react context so I could use it with a raw query client in e.g. route loaders
- for the trpc.useQueries, go for the callback api?